### ISSUE TITLE
Cannot use default VoiceOver keyboard shortcut to open dropdown menu in Firefox/Safari

### ISSUE THREAD

**Issue by 17cliu** [radix-ui/primitives#1963]
## Bug report

When I have VoiceOver enabled, I cannot use VoiceOver's `VO + Space` shortcut to open dropdown menus on Firefox or Safari. (However, it works in Chrome.) I have to use just `Space` (the dropdown's own keyboard handling) to open it.

### Current Behavior

Here's a screenrecording with VoiceOver on Safari, using the demo dropdown menu on [the docs page](https://www.radix-ui.com/docs/primitives/components/dropdown-menu). What I'm doing here:
- I use `VO + Arrow Right` until I'm focused on the dropdown toggle.
- I use `VO + Space` in an attempt to open the menu. Nothing happens.
- At the very end, I use `Space`, which does open the menu.

https://user-images.githubusercontent.com/3460862/219431049-af7c7541-c1b7-4c9a-973a-966e32ceb30e.mov

### Expected behavior

`VO + Space` should open the menu on Firefox and Safari, too. (Not just Chrome!) It's what a VoiceOver user would expect, and that's also what VoiceOver instructs you to press when the dropdown is in focus, so it's very confusing when that shortcut does not work.

### Reproducible example

The dropdown on the docs page is a good example! I was also able to reproduce it in the code base I'm working on, but it's a private repo.

### Suggested solution

**Allow dropdown contents to stay mounted.** (Related to #1273?)

My hunch is that it's not working because the dropdown trigger button does not have the `aria-controls` attribute while the menu is closed. As a result, Firefox and Safari doesn't know which menu the button controls. The `aria-controls` is added to the trigger button only when the dropdown is open, but that's too late. 

Some things I've tried, that didn't alleviate the solution:
- Adding custom ids and a custom `aria-controls` to the dropdown trigger/content. However, the dropdown content is removed, so the id ends up referring to nothing, so this did not work.
- Adding `forceMount` on the content to keep it always rendered. However, this seems to cause the menu to be open at all times? I couldn't figure out a way to keep it mounted and also control its visibility, without affecting any other accessibility features. 

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |    `@radix-ui/react-dropdown-menu`     |    2.0.2     |
| React            | n/a     |         |
| Browser  1      |    Firefox  |      109.0.1    |
| Browser  2      |    Safari    |    16.0    |
| Assistive tech   |    VoiceOver     |         |
| Node             | n/a     |         |
| npm/yarn         |    n/a     |         |
| Operating System |      MacOS  |     12.6 |

Also, I have my VO keys set to `Control+Option` (default), in case that's relevant.

Thank you!

**Comment by bbruneau** [comment]
Hi, I just wanted to chime in and say that I think this is quite important!

**Comment by bbruneau** [comment]
I am not the first to discover this! #2418
